### PR TITLE
Automated script to install Splashkit and native tools on MSYS/MinGW

### PR DESCRIPTION
**Pull request by beastbybirth2**
# Automated script to install Splashkit and native tools on MSYS/MinGW
 
## Description:

This pull request introduces an automated script to streamline the installation process of Splashkit and native development tools on MSYS/MinGW environments. The script offers two options: one for installing Splashkit alone and another for installing Splashkit along with the Visual Studio Code setup.
 
## Changes Made:

- Added an automated script (`script2.bash`) to facilitate the installation process.

- Included two options in the script:

  1. Installation of Splashkit only.

  2. Installation of Splashkit along with Visual Studio Code setup.

- Documented the usage and requirements in the README.md file.
 
## Details:

- **script2.bash**: This script automates the installation process by downloading and setting up Splashkit and necessary native tools on MSYS/MinGW environments. Users are presented with two options:

  1.  script1: Install Splashkit only.

  2. script2: Install Splashkit along with Visual Studio Code setup.
 
- **README.md**: Updated the README file with detailed instructions on how to use the script, system requirements, and additional setup steps for Visual Studio Code integration.
 
To-Do:

- [x] Test the script thoroughly on various MSYS/MinGW environments to ensure compatibility and reliability.

- [x] Address any feedback or suggestions from the community.

- [x] Consider incorporating additional error handling and user prompts for a smoother installation experience.
 
This pull request aims to enhance the accessibility and ease of setting up Splashkit for developers using MSYS/MinGW environments. Feedback and contributions are welcome.

### CODE REVIEW

**Comment by KhushiLaddi08** [comment]
[Car Race Tutorial](https://trello.com/c/YAY93Z4t/592-car-race-tutorial)

File: docs/Splashkit/Applications/mingw-automated installation/script1.bash [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# Install VS Code
+echo "Installing VS Code..."
+curl -sSL https://code.visualstudio.com/sha/download?build=stable&os=win32-x64-user -o vscode.zip
+unzip vscode.zip -d ~/vscode
+echo 'export PATH=$PATH:~/vscode/bin' >> ~/.bashrc
+
+# Install VS Code extensions
+echo "Installing VS Code extensions..."
+code --install-extension ms-vscode.cpptools
```
Comment (omckeon) [inline docs/Splashkit/Applications/mingw-automated installation/script1.bash:43]: Update the extensions to use these:

code --install-extension ms-vscode.cpptools-extension-pack
code --install-extension ms-dotnettools.csharp
code --install-extension ms-dotnettools.csdevkit
code --install-extension ms-dotnettools.vscodeintellicode-csharp

File: docs/Splashkit/Applications/mingw-automated installation/readme.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Installation Script Documentation
```
Comment (omckeon) [inline docs/Splashkit/Applications/mingw-automated installation/readme.md:1]: Remove this file if it does not work correctly

**Review by omckeon — CHANGES_REQUESTED** [review]
I will test the script2.bash file on my windows computer shortly.

File: docs/Splashkit/Applications/mingw-automated installation/script1.bash (hunk 1 above)
Comment (beastbybirth2) [reply docs/Splashkit/Applications/mingw-automated installation/script1.bash]: Done with the required changes.

File: docs/Splashkit/Applications/mingw-automated installation/readme.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+5. **Install VS Code:** Downloads and installs VS Code using PowerShell to handle the installation process.
+6. **Install SplashKit:** Downloads and installs SplashKit using the provided installation script.
+7. **Install VS Code extensions:** Installs essential VS Code extensions for C++ and .NET development.
+
+## Usage
+To use the script, follow these steps:
+
+1. Ensure that you have MSYS2 installed on your system.
+2. Copy the script to your system or download it. Save it as `install.sh` inside your MSYS2 home directory (`C:/msys/home/{username}/install.sh`).
+3. Open a MSYS2 bash terminal window.
```
Comment (omckeon) [inline docs/Splashkit/Applications/mingw-automated installation/readme.md:22]: Which version? MSYS2 or MINGW64?

File: docs/Splashkit/Applications/mingw-automated installation/readme.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+4. **Install .NET SDK:** Downloads and installs the .NET SDK using PowerShell to handle the installation process.
+5. **Install VS Code:** Downloads and installs VS Code using PowerShell to handle the installation process.
+6. **Install SplashKit:** Downloads and installs SplashKit using the provided installation script.
+7. **Install VS Code extensions:** Installs essential VS Code extensions for C++ and .NET development.
+
+## Usage
+To use the script, follow these steps:
+
+1. Ensure that you have MSYS2 installed on your system.
+2. Copy the script to your system or download it. Save it as `install.sh` inside your MSYS2 home directory (`C:/msys/home/{username}/install.sh`).
```
Comment (omckeon) [inline docs/Splashkit/Applications/mingw-automated installation/readme.md:21]: Update to be: `C:/msys64/home/{username}/install.sh`

File: docs/Splashkit/Applications/mingw-automated installation/script2.bash [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+echo "Updating package databases..."
+pacman -Syy
+
+# Install required packages
+echo "Installing required packages..."
+pacman -S --needed --noconfirm git curl unzip
+
+# Install C++ tools (before SplashKit)
+echo "Installing C++ tools..."
+pacman -S --needed --noconfirm mingw-w64-x86_64-gcc mingw-w64-x86_64-make mingw-w64-x86_64-toolchain
```
Comment (omckeon) [inline docs/Splashkit/Applications/mingw-automated installation/script2.bash:18]: Missing gdb.

See this line:
`pacman -S mingw-w64-x86_64-gdb --noconfirm --disable-download-timeout`
from: https://programmers.guide/book/part-0-getting-started/2-computer-use/0-installation/2-6-setup-win-msys/#cc-tools

**Review by omckeon — CHANGES_REQUESTED** [review]
Few typos and some questions. But the script runs successfully with the information from the readme file.

**Comment by omckeon** [comment]
You also need to make sure to run the 4 commands from the CONTRIBUTING.md file (https://github.com/thoth-tech/documentation/blob/main/CONTRIBUTING.md#lint-checks) to ensure the workflows won't fail.

File: docs/Splashkit/Applications/mingw-automated installation/script2.bash
[the remark this answers is not in the document]
Comment (beastbybirth2) [reply docs/Splashkit/Applications/mingw-automated installation/script2.bash:1]: - The file has been renamed.
- The extension has been changed to .sh as all other splashkit scripts use sh.

File: docs/Splashkit/Applications/mingw-automated installation/readme.md (hunk 3 above)
Comment (beastbybirth2) [reply docs/Splashkit/Applications/mingw-automated installation/readme.md]: - It has been updated to MINGW64 in readme.

File: docs/Splashkit/Applications/mingw-automated installation/readme.md (hunk 4 above)
Comment (beastbybirth2) [reply docs/Splashkit/Applications/mingw-automated installation/readme.md]: Changes has been made.

File: docs/Splashkit/Applications/mingw-automated installation/script2.bash (hunk 5 above)
Comment (beastbybirth2) [reply docs/Splashkit/Applications/mingw-automated installation/script2.bash]: - The gdb has been added to installation list.

**Review by omckeon — APPROVED** [review]
Much better :)

**Pull request merged**